test/workbox-cacheable-response/node/CacheableResponse.mjs

```diff
@@ -0,0 +1,162 @@
+      await expectError(() => {
+        new CacheableResponse({statuses: 'bad input'});
+      }, 'not-an-array', (err) => {
+        expect(err.details).to.have.property('moduleName').that.eql('workbox-cacheable-response');
+        expect(err.details).to.have.property('className').that.eql('CacheableResponse');
+        expect(err.details).to.have.property('funcName').that.eql('constructor');
+        expect(err.details).to.have.property('paramName').that.eql('config.statuses');
+      });
+    });
+
+    devOnly.it(`should throw with bad config.headers`, async function() {
+      await expectError(() => {
+        new CacheableResponse({headers: 'bad input'});
+      }, 'incorrect-type', (err) => {
+        expect(err.details).to.have.property('moduleName').that.eql('workbox-cacheable-response');
+        expect(err.details).to.have.property('className').that.eql('CacheableResponse');
+        expect(err.details).to.have.property('funcName').that.eql('constructor');
+        expect(err.details).to.have.property('paramName').that.eql('config.headers');
+      });
+    });
+
+    it(`should be able to construct with config.statuses`, function() {
+      const cacheableResponse = new CacheableResponse({statuses: VALID_STATUSES});
+
+      expect(cacheableResponse._statuses).to.eql(VALID_STATUSES);
+    });
+
+    it(`should be able to construct with config.headers`, function() {
+      const cacheableResponse = new CacheableResponse({headers: VALID_HEADERS});
+
+      expect(cacheableResponse._headers).to.eql(VALID_HEADERS);
+    });
+
+    it(`should be able to construct with config.statuses and config.headers`, function() {
+      const cacheableResponse = new CacheableResponse({
+        statuses: VALID_STATUSES,
+        headers: VALID_HEADERS,
+      });
+
+      expect(cacheableResponse._statuses).to.eql(VALID_STATUSES);
+      expect(cacheableResponse._headers).to.eql(VALID_HEADERS);
+    });
+  });
+
+  describe(`isResponseCacheable`, function() {
+    devOnly.it(`should throw when passed bad input`, async function() {
+      const cacheableResponse = new CacheableResponse({statuses: VALID_STATUSES});
+      await expectError(() => {
+        cacheableResponse.isResponseCacheable(null);
+      }, 'incorrect-class', (err) => {
+        expect(err.details).to.have.property('moduleName').that.eql('workbox-cacheable-response');
+        expect(err.details).to.have.property('className').that.eql('CacheableResponse');
+        expect(err.details).to.have.property('funcName').that.eql('isResponseCacheable');
+        expect(err.details).to.have.property('paramName').that.eql('response');
+      });
+    });
+
+    it(`should return true when one of the statuses match the response`, function() {
+      const cacheableResponse = new CacheableResponse({statuses: VALID_STATUSES});
+      const response = new Response('', {status: VALID_STATUS});
+      // TODO: Remove after there's a proper Response mock.
```

Comment:
There are incompatibilities with the `.headers` implementation in the default `Response` mock provided by `service-worker-mocks`.

I was going to resolve those incompatibilities by switching over to our own `Response` mock as part of this PR, but unfortunately we have other tests and some module code that relies on the incorrect behavior.

Stay tuned for a separate PR to resolve those issues. Once that's in place, I can take out this manual override.

Reply:
https://github.com/GoogleChrome/workbox/pull/1059